backend/src/case/case.service.ts

```diff
@@ -1,70 +1,2304 @@
+    private readonly authHelperService: AuthHelperService,
   ) {}
 
-  async createCase(createCaseDTO: CreateCaseDto, userId: string) {
+  async createCaseSystemTransmission(payload: SystemCaseCreationDto, clientId: string, tenantId: string) {
     try {
-      this.logger.log('Creating case', CaseService.name);
+      this.logger.log('System-to-system case creation initiated', CaseService.name);
+      const systemUuid = this.configService.get<string>('SYSTEM_UUID', clientId);
 
-      const createdCase = await this.prismaService.case.create({
+      this.eventEmitter.emit('alert.incoming', {
+        payload,
+        source: 'REST API',
+        userId: systemUuid,
+        tenantId,
+      });
+
+      await this.auditLogService.logAction({
+        userId: systemUuid,
+        operation: 'createCase',
+        entityName: CaseService.name,
+        actionPerformed: 'Case creation triggered via system transmission',
+        outcome: Outcome.SUCCESS,
+      });
+      return { message: 'Case creation triggered via system transmission' };
+    } catch (error) {
+      this.logger.error(`Error in system-to-system case creation: ${error.message}`, error.stack, CaseService.name);
+      throw error;
+    }
+  }
+
+  async manualCaseCreate(dto: ManualCreateCaseDto, userId: string, tenantId: string, role: string) {
+    this.logger.log(`[ManualCase] Starting manual case creation by user ${userId} with role ${role}`, CaseService.name);
+
+    if (!dto.alertId || !dto.alertType) {
+      this.logger.error('[ManualCase] Missing required fields in ManualCreateCaseDto', '', CaseService.name);
+      throw new BadRequestException('alertId and alertType are required');
+    }
+
+    const existingAlert = await this.prismaService.alert.findUnique({
+      where: { alert_id: dto.alertId },
+    });
+
+    if (!existingAlert) {
+      throw new NotFoundException(`Alert ${dto.alertId} not found`);
+    }
+
+    if (existingAlert.case_id) {
+      this.logger.error(`[ManualCase] Case already exists for alertId ${dto.alertId}`, '', CaseService.name);
+      throw new BadRequestException(`Case already exists for alertId ${dto.alertId}`);
+    }
+
+    const alertStatus = (existingAlert.alert_data as any)?.status;
+    if (alertStatus !== 'NALT') {
+      this.logger.error('[ManualCase] Cannot create Case: alert_data.status is not NALT', '', CaseService.name);
+      throw new BadRequestException('Cannot create Case: alert_data.status is not NALT');
+    }
+
+    const priorityScore = dto.priorityScore ?? 0.33;
+    const priority = this.casePriorityUtil.determinePriority(priorityScore);
+    const caseType = this.casePriorityUtil.mapAlertTypeToCaseType(dto.alertType);
+
+    const needsApproval = role !== 'SUPERVISOR';
+    const caseStatus = needsApproval ? CaseStatus.STATUS_01_PENDING_CASE_CREATION_APPROVAL : CaseStatus.STATUS_02_READY_FOR_ASSIGNMENT;
+    const caseOwnerId = needsApproval ? undefined : userId;
```

Comment:
Maybe move the constant creation to the start. And change the condition like so. Avoids multiple ternary operators.

```suggestion
    const needsApproval = role !== 'SUPERVISOR';
    const caseStatus;
    const caseOwnerId;
    
        if (needsApproval) {
      caseStatus = CaseStatus.STATUS_01_PENDING_CASE_CREATION_APPROVAL;
      caseOwnerId = null;
    } else {
      caseStatus = CaseStatus.STATUS_02_READY_FOR_ASSIGNMENT;
      caseOwnerId = userId;
    }
```